action_plugins/play_sound/PlaySoundAction.qml

```diff
@@ -0,0 +1,94 @@
+// -*- coding: utf-8; -*-
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+import QtQuick.Window
+import QtQuick.Dialogs
+
+import QtQuick.Controls.Universal
+
+import Gremlin.ActionPlugins
+import "../../qml"
+
+
+Item {
+    property PlaySoundModel action
+
+    implicitHeight: _content.height
+
+    RowLayout {
+        id: _content
+
+        anchors.left: parent.left
+        anchors.right: parent.right
+
+        Label {
+            id: _label
+
+            Layout.preferredWidth: 110
+
+            text: "Audio filename"
+        }
+
+        TextField {
+            id: _soundFilename
+
+            Layout.fillWidth: true
+
+            placeholderText: null != action ? null : "Input the name of the audio file to play."
+            text: action.sound_filename
+            selectByMouse: true
+
+            onTextChanged: {
+                action.sound_filename = text
+            }
+        }
+
+        Button {
+            text: "Select File"
+            onClicked: _fileDialog.open()
+        }
+
+        Label {
+            Layout.preferredWidth: 50
+            text: "Volume"
+        }
+
+        TextField {
```

Comment:
I would use a SpinBox (https://doc.qt.io/qt-6/qml-qtquick-controls-spinbox.html) here as the input is meant to be an integer. That widget lets you set minimum and maximum values. The `MapToMouseAction.qml` file is a decent example of how to use that widget.

Reply:
Yup, no issues.